File: frontend/lib/domains/inbound/register_dto_in.dart
import '/domains/gender.dart';

class RegisterDtoIn {
  String? firstName;
  String? lastName;
  String? email;
  String? phoneNumber;
  Gender? gender;
  String? password;

  RegisterDtoIn({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.phoneNumber,
    required this.gender,
    required this.password,
  });

  factory RegisterDtoIn.fromJson(Map<String, dynamic> json) {
    return RegisterDtoIn(
      firstName: json['firstName'] as String?,
      lastName: json['lastName'] as String?,
      email: json['email'] as String?,
      phoneNumber: json['phoneNumber'] as String?,
      gender: json['gender'] != null
          ? Gender.values.byName(json['gender'] as String)
          : null,
      password: json['password'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'phoneNumber': phoneNumber,
      'gender': gender?.name,
      'password': password,
    };
  }
}

File: backend/src/main/java/com/etekhno/rentacar/domain/inbound/RegisterDTOIn.java
package com.etekhno.rentacar.domain.inbound;

import com.etekhno.rentacar.datamodel.enums.Gender;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class RegisterDTOIn {
    @NotNull
    @NotBlank
    private String firstName;
    @NotNull
    @NotBlank
    private String lastName;
    @Email
    @NotNull
    @NotBlank
    private String email;
    private String cnic;
    @NotBlank
    @NotNull
    private String phoneNumber;
    @NotNull
    private Gender gender;
    @NotBlank
    @NotNull
    @Pattern(regexp = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&-+=()])(?=\\S+$).{8,20}$",
            message = "should have 8 characters, contain at least one small and capital letter and special symbol")
    private String password;
}
